packages/contracts/test/exchange/helpers.ts

```diff
@@ -1,52 +1,68 @@
 import { ZeroEx } from '0x.js';
+import { BlockchainLifecycle, devConstants, web3Factory } from '@0xproject/dev-utils';
 import { BigNumber } from '@0xproject/utils';
+import { Web3Wrapper } from '@0xproject/web3-wrapper';
 import * as chai from 'chai';
 import ethUtil = require('ethereumjs-util');
 
-import { Artifacts } from '../../../util/artifacts';
-import { ExchangeWrapper } from '../../../util/exchange_wrapper';
-import { Order } from '../../../util/order';
-import { OrderFactory } from '../../../util/order_factory';
+import { constants } from '../../util/constants';
+import { ExchangeWrapper } from '../../util/exchange_wrapper';
+import { Order } from '../../util/order';
+import { OrderFactory } from '../../util/order_factory';
 import { chaiSetup } from '../utils/chai_setup';
+import { deployer } from '../utils/deployer';
 
 chaiSetup.configure();
 const expect = chai.expect;
 
-const { Exchange, TokenRegistry } = new Artifacts(artifacts);
+const web3 = web3Factory.create();
+const web3Wrapper = new Web3Wrapper(web3.currentProvider);
+const blockchainLifecycle = new BlockchainLifecycle(devConstants.RPC_URL);
 
-contract('Exchange', (accounts: string[]) => {
-    const maker = accounts[0];
-    const feeRecipient = accounts[1] || accounts[accounts.length - 1];
+describe('Exchange', () => {
+    let maker: string;
+    let feeRecipient: string;
 
     let order: Order;
     let exchangeWrapper: ExchangeWrapper;
     let orderFactory: OrderFactory;
 
     before(async () => {
-        const [tokenRegistry, exchange] = await Promise.all([TokenRegistry.deployed(), Exchange.deployed()]);
-        exchangeWrapper = new ExchangeWrapper(exchange);
-        const [repAddress, dgdAddress] = await Promise.all([
-            tokenRegistry.getTokenAddressBySymbol('REP'),
-            tokenRegistry.getTokenAddressBySymbol('DGD'),
+        const accounts = await web3Wrapper.getAvailableAddressesAsync();
+        maker = accounts[0];
+        feeRecipient = accounts[1] || accounts[accounts.length - 1];
```

Comment:
What is this magic? In which scenarios is `accounts[1]` undefined?

Reply:
I wish I knew the answer. It was like that before.